bladerf fails to build

**Describe the bug**

libbladeRF fails to build on the nixos-unstable channel and latest master:
```
/build/source/host/libraries/libbladeRF/src/board/bladerf1/bladerf1.c: In function 'bladerf1_open':
/build/source/host/libraries/libbladeRF/src/board/bladerf1/bladerf1.c:841:38: error: implicit conversion from 'enum <anonymous>' to 'enum <anonymous>' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wenum-conversion-Werror=enum-conversion8;;]
  841 |     dev->flash_arch->status          = STATE_UNINITIALIZED;
      |                                      ^
``` 

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@Mic92 @pSub

--- a/pkgs/development/libraries/libbladeRF/default.nix
+++ b/pkgs/development/libraries/libbladeRF/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, lib, fetchFromGitHub, pkgconfig, cmake, git, doxygen, help2man, ncurses, tecla
+{ stdenv, lib, fetchFromGitHub, fetchpatch, pkgconfig, cmake, git, doxygen, help2man, ncurses, tecla
 , libusb1, udev }:
 
 let
@@ -22,6 +22,15 @@ in stdenv.mkDerivation {
     rev = "libbladeRF_v${version}";
     sha256 = "0g89al4kwfbx1l3zjddgb9ay4mhr7zk0ndchca3sm1vq2j47nf4l";
   };
+
+  # This patch is required for version 2.2.1. As the patch is already part of
+  # upstream master, it will be incorporated into the next release. The patch
+  # fixes a (well-justified) compiler warning which breaks the build because
+  # we compile with -Werror.
+  patches = [ (fetchpatch {
+    url = "https://github.com/Nuand/bladeRF/commit/163425d48a3b7d8c100d7295220d3648c050d0dd.patch";
+    sha256 = "1swsymlyxm3yk2k8l71z1fv0a5k2rmab02f0c7xkrvk683mq6yxw";
+  }) ];
 
   nativeBuildInputs = [ pkgconfig ];
   # ncurses used due to https://github.com/Nuand/bladeRF/blob/ab4fc672c8bab4f8be34e8917d3f241b1d52d0b8/host/utilities/bladeRF-cli/CMakeLists.txt#L208

